File: repos/firecrawl/apps/java-sdk/src/main/java/com/firecrawl/errors/JobTimeoutException.java
package com.firecrawl.errors;

/**
 * Thrown when an async job (crawl, batch, agent) does not complete within the specified timeout.
 */
public class JobTimeoutException extends FirecrawlException {

    private final String jobId;
    private final int timeoutSeconds;

    public JobTimeoutException(String jobId, int timeoutSeconds, String jobType) {
        super(jobType + " job " + jobId + " did not complete within " + timeoutSeconds + " seconds");
        this.jobId = jobId;
        this.timeoutSeconds = timeoutSeconds;
    }

    /** The ID of the timed-out job. */
    public String getJobId() { return jobId; }

    /** The timeout in seconds that was exceeded. */
    public int getTimeoutSeconds() { return timeoutSeconds; }
}

File: repos/firecrawl/apps/dot-net-sdk/Firecrawl/Exceptions/JobTimeoutException.cs
namespace Firecrawl.Exceptions;

/// <summary>
/// Thrown when an async job (crawl, batch scrape, agent) does not complete within the specified timeout.
/// </summary>
public class JobTimeoutException : FirecrawlException
{
    /// <summary>
    /// The ID of the job that timed out.
    /// </summary>
    public string JobId { get; }

    /// <summary>
    /// The timeout in seconds that was exceeded.
    /// </summary>
    public int TimeoutSeconds { get; }

    public JobTimeoutException(string jobId, int timeoutSeconds, string jobType)
        : base($"{jobType} job {jobId} did not complete within {timeoutSeconds} seconds")
    {
        JobId = jobId;
        TimeoutSeconds = timeoutSeconds;
    }
}
